### ISSUE TITLE
When connection crashes, can no longer use that subdomain

### ISSUE THREAD

**Issue by drewandre** [localtunnel/localtunnel#220]
I understand that good work is being done on this library to help improve reliability. I wanted to bring up another issue that may be related in an adjacent way. I'm also curious if anyone else is having this issue.

My connection frequently disconnects, but I then can't use that same subdomain... I don't mind having to restart localtunnel every now and then, but losing access to that subdomain is pretty frustrating.

For example, I use _mydomain_ as the subdomain on port 3000, it crashes, and when I run `lt --subdomain mydomain --port 3000` again, it generates the random characters. I then have to use something like _mydomain1_, and update the server address on every single sensor I have (which send data over a cellular connection to my rails server), which is definitely frustrating.

Has anyone else had this issue?

Is there any way to kill all instances of localtunnel if there are instances running...?

**Comment by netfaruk** [comment]
I have same issue. I haven't found a solution yet.

**Comment by Alvarito050506** [comment]
I also have that problem.

Sorry for my english, I am an argentine boy.

**Comment by jpsear** [comment]
Yep, same here. Would be great to have a `--close` flag to use on the command line...

**Comment by usamaejaz** [comment]
I am also having this same issue.

**Comment by highfeed** [comment]
I have same issue.

**Comment by rishabhp16** [comment]
Yeah, I'm also facing the same issue

**Comment by highfeed** [comment]
I have this problem again.

**Comment by 2tonewarrior** [comment]
Same  problem here :(

**Comment by onurtskn** [comment]
Still occurs. BTW after some time, the subdomain becomes available again.

**Comment by Alvarito050506** [comment]
Yes, but after 24hs.

**Comment by eric-hf** [comment]
Is there any way to see _when_ the connection crashed? That way you could know when the subdomain will be freed up by the 24h reset.

**Comment by RoboticMind** [comment]
@defunctzombie Can you take a look at this?

**Comment by defunctzombie** [comment]
Will look into this; restarting with a given subdomain is a supported feature so if there is a bug it will need to be diagnosed. If there are steps to reproduce that would be helpful.

**Comment by defunctzombie** [comment]
x-post from #227 

> I suspect in your case you are hitting the grace period of subdomains. A subdomain is not handed back out immediately after all tunnel connections to it are dropped to allow the client that created it to reconnect; however if you exit the client and start it again it will no longer remember the internal connectivity information and you will not be able to re-use the same subdomain for a short period. I'll think about dropping the grace period altogether as there is unlikely to be sniping occurring :)

**Comment by RoboticMind** [comment]
Also for the Reproducing the bug: 
1) keep a connection open for a very long time and every hour or so, ping it and see if it times out
2) when the connection times out it won't start work working again so close the connection
3) try to get the same subdomain again and you won't be able to

**Comment by defunctzombie** [comment]
@RoboticMind few questions:

* What does it mean to close the connection? exit localtunnel client cli?
* How quickly do you try to get the same subdomain? Is there a period after which you are able to get the same subdomain?

**Comment by RoboticMind** [comment]
When I said close the connection, I meant to say close the program in the terminal via pressing ctrl-c. The subdomain goes unavailable for at least 16+ hours (I haven't tested it very rigorously).

**Comment by slorenzo** [comment]
@defunctzombie  I use `nodemon` to automatically restart my node server and, sometimes, it also happens when I make some changes in my application.

Localtunnel logs:
`/usr/local/lib/node_modules/localtunnel/bin/client:58`
 `       throw err;`
`        ^`
`Error: connection refused: localtunnel.me:36312 (check your firewall settings)`
`    at Socket.<anonymous> (/usr/local/lib/node_modules/localtunnel/lib/TunnelCluster.js:47:32)`
`    at Socket.emit (events.js:127:13)`
`    at emitErrorNT (internal/streams/destroy.js:64:8)`
`    at process._tickCallback (internal/process/next_tick.js:114:19)`

Then, I try to restart localtunnel but it provide me a different URL and ignores my subdomain.

**Comment by bobbytables** [comment]
I'm seeing this too, is there a plan to drop the grace period?

**Comment by defunctzombie** [comment]
I've deployed a new server backend that should resolve this issue. If your client is disconnected (or dies) then the subdomain will be available again within a second.

**Comment by Sangdol** [comment]
@defunctzombie thanks for the work. which version is it?

**Comment by abhinaykumar** [comment]
I am facing the same issue even after the patch.

**Comment by CyberQuestor** [comment]
Just blocked two sub-domains again. Occurs more frequently when I am on VPN trying to access network resources in parallel. Suspect a lingering connection after exiting application (ctrl +c).

Easiest way to reproduce:
- Open a tunnel `lt -s <custom_subdomain> --port <port>`
- Once you receive desired sub-domain, disconnect wi-fi/ network adapter (to emulate possible connection drops)
- Exit application (ctrl + c)
- Connect back wi-fi/ network adapter
- Attempt to open tunnel with same sub-domain `lt -s <custom_subdomain> --port <port>`
- You will no longer be able to claim requested sub-domain

Failed to locate any mechanism to forcefully emit a `reset`/ `dead` or `close` request to trigger `sock.destroy();`. I suppose its back to you @defunctzombie .

**Comment by BerndErnst** [comment]
How about to reopen this issue?
I think @abhinaykumar and @drewandre  already got some likes (also from me)

**Comment by defunctzombie** [comment]
https://github.com/localtunnel/localtunnel/issues/248 This issue is open and tracks the same problem @BerndErnst

**Comment by asoap** [comment]
I am having this same issue as well.  I'm developing an app that connects directly to a hot spot.  And I need to test the wifi going offline.  Hence I'm turning my router hot spot off, which then doesn't let me reconnect to local tunnel for a long period of time.

**Comment by bellaj** [comment]
Same here, any solution in sight?

**Comment by ADavidLiu** [comment]
I'm still having the same issue.
